src/openforms/formio/migration_converters.py

```diff
@@ -127,6 +127,22 @@ def fix_file_default_value(component: Component) -> bool:
             return False
 
 
+def ensure_extra_zip_mimetypes_exist_in_file_type(component: Component) -> bool:
+    component = cast(FileComponent, component)
+    if not (file_type := glom(component, "file.type", default=None)):
+        return False
+
+    if "application/zip" in file_type:
+        if "application/x-zip-compressed" not in file_type:
+            file_type.append("application/x-zip-compressed")
+        if "application/zip-compressed" not in file_type:
+            file_type.append("application/zip-compressed")
+            "application/zip-compressed"
+
+        assign(component, "file.type", file_type)
+        return True
```

Comment:
I'm not sure this is sufficient, since there's also the `filePattern` key which is sort of derived? See https://github.com/open-formulieren/open-forms/blob/ea5282deac7704e1d6d6f57b13fcc7aa0ad7b558/src/openforms/formio/components/vanilla.py#L413